exec/internal/compile/scanner.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package compile
+
+import (
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+)
+
+type scanner struct {
+	supportedOpcodes map[byte]bool
+}
+
+// InstructionMetadata describes a bytecode instruction.
+type InstructionMetadata struct {
+	Op    byte
+	Start int
```

Comment:
we should probably document what `Start` represents, where it starts and what are the expected units.